functions/go/enable-gcp-services/README.md

```diff
@@ -0,0 +1,120 @@
+# enable-gcp-services
+
+## Overview
+
+<!--mdtogo:Short-->
+
+The `enable-gcp-services` function generates [GCP project service](https://cloud.google.com/config-connector/docs/reference/resource-docs/serviceusage/service)
+resources from a list of services to enable [GCP APIs](https://cloud.google.com/apis) within a specified project. This allows users to succinctly define all
+the services necessary in a single resource and have tighter control over which services are enabled in a specific project.
+
+<!--mdtogo-->
+
+<!--mdtogo:Long-->
+
+## Usage
+
+`enable-gcp-services` function can be used both declaratively and imperatively.
+
+```shell
+kpt fn eval --image gcr.io/kpt-fn/enable-gcp-services:unstable
+```
+
+The `enable-gcp-services` function does the following:
+
+1. Discovers all `ProjectServiceList` custom resources in a given package.
+
+1. For each `ProjectServiceList` CR, it generates [GCP project service](https://cloud.google.com/config-connector/docs/reference/resource-docs/serviceusage/service) resources as specified in the `spec.services` list.
+    * Adds all annotations defined for `ProjectServiceList` CR to each generated resource. This can be used for enabling features like
+[disable-on-destroy](https://cloud.google.com/config-connector/docs/reference/resource-docs/serviceusage/service#custom_resource_definition_properties) for generated services.
+    * Sets namespace if any defined for `ProjectServiceList` CR to each generated resource.
+    * Sets projectID if any defined for `ProjectServiceList` CR to each generated resource.
+1. Each generated [GCP project service](https://cloud.google.com/config-connector/docs/reference/resource-docs/serviceusage/service) resource
+has a `blueprints.cloud.google.com/managed-by-enable-gcp-services` annotation. This annotation allows `enable-gcp-services` function to
+track generated resources for the declarative management of the generated resources. Any changes made to the generate resources will be overwritten and should be made to the `ProjectServiceList` CR instead.
+
+### `ProjectServiceList`
```

Comment:
@mengqiy would `EnableGCPServices` be better? /cc @morgante if you have any thoughts.

Reply:
CRDs should be nouns, not verbs.

Since `List` is a reserved name, how about `ProjectServiceSet`. There seems to be some precedent for using `Set` for non-list collections. Ex. https://docs.projectcalico.org/reference/resources/globalnetworkset

Reply:
@mengqiy can we move ahead with `ProjectServiceSet`?